### PR TITLE
HRJS-36 SASL Authentication

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/HRJS-36

This one adds `DIGEST-MD5`, `SCRAM-SHA-1`, `SCRAM-SHA-256`, `SCRAM-SHA-384`, `SCRAM-SHA-512` and `EXTERNAL` mechanisms (the latter without a test for now).

### CODE REVIEW

File: run-testsuite.sh [hunk 1]
```diff
@@ -3,4 +3,5 @@
 # Stop if any errors found
 set -e
 
-node --trace-deprecation node_modules/jasmine-node/lib/jasmine-node/cli.js spec/* --captureExceptions
```
Comment (karesti) [inline run-testsuite.sh:6]: needs to be rollbacked (modified for quick run only)

File: run-servers.sh [hunk 2]
```diff
@@ -117,21 +117,21 @@ rm -drf server/${SERVER_DIR}
...
-prepareServerDir "$1" infinispan-clustered.xml false "server-failover-one"
-prepareServerDir "$1" infinispan-clustered.xml false "server-failover-two"
-prepareServerDir "$1" infinispan-clustered.xml false "server-failover-three"
-
-#Preparing server dirs for xsite tests (2 servers)
-prepareServerDir "$1" infinispan-xsite-EARTH.xml false "server-earth"
-prepareServerDir "$1" infinispan-xsite-MOON.xml false "server-moon"
-
-waitForClusters
+#startServer "$1" infinispan-clustered.xml false 11322 "server-one"
```
Comment (karesti) [inline run-servers.sh:120]: needs to be rollbacked to run all the tests

**Review by karesti — CHANGES_REQUESTED** [review]
I commented stuff to run faster some stuff and tests but need to be rollbacked. otherwise LGTM!

**Comment by karesti** [comment]
@tristantarrant needs rebase for the run-servers. I can do it

**Review by karesti — APPROVED** [review]

**Comment by tristantarrant** [comment]
I've added docs to the README and the jsdoc

**Review by karesti — APPROVED** [review]

**Pull request merged**